scratch-js/Sprite.mjs

```diff
@@ -95,46 +95,57 @@ class SpriteBase {
 
   set costumeNumber(number) {
     this._costumeNumber = ((number - 1) % this.costumes.length) + 1;
+    this.fireBackdropTrigger();
+  }
+
+  fireBackdropTrigger() {
+    if (this instanceof Stage) {
+      this._project.fireTrigger(Trigger.BACKDROP_CHANGED, {
+        backdrop: this.costume.name
+      });
+    }
   }
```

Comment:
Might it make more sense for this method to exist only on the `Stage` class? Then the `if (this instanceof Stage)` check could be placed inside of `set costumeNumber()`.